fe/src/main/java/org/apache/doris/mysql/privilege/UserPrivTable.java

```diff
@@ -141,37 +146,37 @@ public boolean checkPlainPassword(String remoteUser, String remoteHost, String r
             }
 
             if (MysqlPassword.checkPlainPass(globalPrivEntry.getPassword(), remotePasswd)) {
+                if (currentUser != null) {
+                    if (entry.isSetByDomainResolver()) {
+                        currentUser.add(globalPrivEntry.getDomainUserIdent());
+                    } else {
+                        currentUser.add(entry.getUserIdent());
+                    }
```

Comment:
Could you encapsulate this logic in getDomainUserIdent(), then it would be not everywhere to judge this condition

Reply:
OK